homeassistant/components/arcam_fmj/__init__.py

```diff
@@ -122,22 +142,23 @@ async def async_setup_entry(hass: HomeAssistantType, entry: config_entries.Confi
     return True
 
 
-async def _run_client(hass, client, interval):
-    task = asyncio.Task.current_task()
-    run = True
+async def async_unload_entry(hass, entry):
+    """Cleanup before removing config entry."""
+    await hass.config_entries.async_forward_entry_unload(entry, "media_player")
 
-    async def _stop(_):
-        nonlocal run
-        run = False
-        task.cancel()
-        await task
+    task = hass.data[DOMAIN_DATA_TASKS].pop(entry.entry_id)
+    if task:
```

Comment:
When is this not True?

Reply:
Probably never. Just defensive coding. But I agree, no point in protecting

Reply:
I've resolved this, but I just realised that if you add typing tot his i'd have to re-add it (or cast) since pop would be returning a Optional[Any]. Oh well :)

Reply:
There's no default for pop unless specifically added. It will raise if the key is missing without default value.